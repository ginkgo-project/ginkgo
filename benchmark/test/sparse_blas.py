#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-operations", "transpose", "-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="sparse_blas.simple.stdout",
    expected_stderr="sparse_blas.simple.stderr",
)

# stdin
test_framework.compare_output(
    ["-operations", "transpose"],
    expected_stdout="sparse_blas.simple.stdout",
    expected_stderr="sparse_blas.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    [
        "-operations",
        "transpose",
        "-input",
        str(test_framework.sourcepath / "input.mtx.json"),
    ],
    expected_stdout="sparse_blas.simple.stdout",
    expected_stderr="sparse_blas.simple.stderr",
)

# profiler annotations (transpose has the smallest number of allocations)
test_framework.compare_output(
    [
        "-operations",
        "transpose",
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="sparse_blas.profile.stdout",
    expected_stderr="sparse_blas.profile.stderr",
)
