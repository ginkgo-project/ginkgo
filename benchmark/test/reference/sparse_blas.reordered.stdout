[
    {
        "size": 100,
        "stencil": "7pt",
        "sparse_blas": {
            "symbolic_cholesky": {
                "time": 1.0,
                "flops": 1.0,
                "bandwidth": 1.0,
                "repetitions": 10,
                "components": {
                    "compute_elimination_forest": 1.0,
                    "allocate": 1.0,
                    "free": 1.0,
                    "components::fill_array": 1.0,
                    "cholesky::symbolic_count": 1.0,
                    "components::prefix_sum_nonnegative": 1.0,
                    "copy": 1.0,
                    "cholesky::symbolic_factorize": 1.0,
                    "csr::sort_by_column_index": 1.0,
                    "overhead": 1.0
                },
                "factor_nonzeros": 1324,
                "completed": true
            }
        },
        "reordered": "amd",
        "rows": 125,
        "cols": 125,
        "nonzeros": 725
    }
]
